pkg/mqtt-paho/contrib/riot_iface.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright (C) 2019  Javier FILEIV <javier.fileiv@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @brief       MQTT common RIOT interface functions
+ *
+ * @author      Javier FILEIV <javier.fileiv@gmail.com>
+ */
+#include <string.h>
+#include <net/ipv6/addr.h>
+#include <net/af.h>
+#include <net/sock.h>
+#include <net/sock/tcp.h>
+#include <mqtt.h>
+#include <log.h>
+
+#ifndef MQTT_THREAD_PRIORITY
+#define MQTT_THREAD_PRIORITY    (THREAD_PRIORITY_MAIN - 4)
```

Comment:
This doesn't work at all. On native, I have a segmentation fault when connecting to the broker. On a samr21-xpro board, setup behind a border router, the application crashes immediately when connecting.
Using `THREAD_PRIORITY_MAIN + 1` works on native but still crashes on a board.